setup.py

```diff
@@ -8,27 +8,21 @@ def load_requires_from_file(filepath):
     return [pkg_name.rstrip('\r\n') for pkg_name in open(filepath).readlines()]
 
 
-def install_requires():
-    requires = load_requires_from_file('requirements.txt')
-    if sys.version_info < (3, 4):
-        # To enable Enum in Python < 3.4
-        requires.append('enum34 >= 1.0.4')
-        # To enable pathlib in Python < 3.4
-        requires.append('pathlib == 1.0.1')
+install_requires = load_requires_from_file('requirements.txt')
+install_requires += [
+    'setuptools>=36.2.2',  # for enhanced marker support (used below).
+    'enum34>=1.0.4;python_version<"3.4"',
+    'pathlib==1.0.1;python_version<"3.4"',
+    'typing>=3.6.2;python_version<"3.6"',
+]
```

Comment:
👍

##### NOTE
I have not knew this specification, but I found it in [PEP-426](https://legacy.python.org/dev/peps/pep-0426/#environment-markers). And this specification was introduced since [0.7](https://pypi.org/project/setuptools/0.7.3/#id4) (2013), so I feel it is a widely usable feature.

Reply:
Re `pathlib`: should become `>=` probably.